File: HourRank29_P1.cpp
// code_report Solution
// https://youtu.be/YHoEJ_UhG1Y

#include <vector>
#include <string>

using namespace std;

#define FORI(s,n) for(int i = s; i < n; i++)
#define FORJ(s,n) for(int j = s; j < n; j++)

string solve (vector<vector<int>> b) 
{
   int n = b.size ();
   FORI (0, n) FORJ (1, n) if (b[i][j] == b[i][j - 1]) return "No";
   FORI (0, n) FORJ (1, n) if (b[j][i] == b[j - 1][i]) return "No";
   return "Yes";
}

File: HourRank29_P1.java
// code_report Solution
// https://youtu.be/YHoEJ_UhG1Y

static String solve(List<List<Integer>> b) 
{
    int n = b.size ();
    for (int i = 0; i < n; ++i) {
        for (int j = 1; j < n; ++j) {
            if (b.get (i).get (j) == b.get (i).get (j - 1)) return "No";
            if (b.get (j).get (i) == b.get (j - 1).get (i)) return "No";
        }
    }
    return "Yes";
}
